### DISCUSSION TITLE
Deploying Ubuntu Desktop 18.04 LTS via http (without using NFS)

### DISCUSSION THREAD

**Discussion by jamaya77** [ipxe/ipxe#319] [Q&A]
Hi,

This isn't an iPXE issue, but I'm just asking the community for any suggestions for deploying Ubuntu 18.04 Desktop ISO over http, without using NFS. I currently have a similar [solution](https://gist.github.com/tlhakhan/03dbb4867f70d17d205c179a58fd5923) for deploying Ubuntu 20.04 which seems to work great, but it won't work with 18.04. Thoughts?

#!ipxe

#$seedfrom used to find the user-data and meta-data files by nocloud-net provider for cloud-init.
#the trailing slash is important, the cloud-init sticks 'meta-data' or 'user-data' right after it, without prepending a forward slash.
set seedfrom http://192.168.140.21/iso/ubuntu1804

#$base url is where the vmlinuz and initrd live.
#they were fished out from the live-server iso file. the iso file is also in this directory.
set base http://192.168.140.21/iso/ubuntu1804/

kernel ${base}/vmlinuz initrd=initrd.lz autoinstall url=${base}/ubuntu1804.iso net.ifnames=0 ip=dhcp ipv6.disable=1 ds=nocloud-net;s=${seedfrom}
initrd ${base}/initrd.lz
boot

<img width="994" alt="1804http" src="https://user-images.githubusercontent.com/80473349/114327282-687c0780-9b06-11eb-9aa0-42b4f5f0f841.png">

**Comment by ondrejj** [comment]
May be you should change initrd.lz to initrd.gz (older version of compression for older distribution).
If don't try to look at http://boot.salstar.sk/deb.ipxe, which contains a working example.

**Reply by jamaya77** [reply]
Thank you @ondrejj for your suggestions. I renamed the file from .lz to .gz, but nothing changed, Still get the same error.
I'll take a look at the script you advised, but it looks complicated.

**Comment by sebaxakerhtc** [comment]
> set base http://192.168.140.21/iso/ubuntu1804/

> url=${base}/ubuntu1804.iso

Seems your link looks like this:
`http://192.168.140.21/iso/ubuntu1804//ubuntu1804.iso`
You use two slashes!
The problem, i think, here.

Possible solutions are:
change here `set base http://192.168.140.21/iso/ubuntu1804`
or change here `url=${base}ubuntu1804.iso`

**Comment by sebaxakerhtc** [comment]
> I currently have a similar [solution](https://gist.github.com/tlhakhan/03dbb4867f70d17d205c179a58fd5923) for deploying Ubuntu 20.04

Your mistake at this point, because this example use `set seedfrom link/to/ubuntu2004/`with slash in the end, but yours is not
and `set base link/to/ubuntu2004` without slash in the end, but yours with it

**Reply by jamaya77** [reply]
Thank you @sebaxakerhtc for your suggestion.
I've made the modified the script, but I still receive the same results as the original snapshot above:

#!ipxe
#ubuntu 18.04

#$seedfrom used to find the user-data and meta-data files by nocloud-net provider for cloud-init.
#the trailing slash is important, the cloud-init sticks 'meta-data' or 'user-data' right after it, without prepending aforward slash.
set seedfrom http://192.168.140.21/iso/ubuntu1804

#$base url is where the vmlinuz and initrd live.
#they were fished out from the live-server iso file. the iso file is also in this directory.
set base http://192.168.140.21/iso/ubuntu1804

kernel ${base}/vmlinuz initrd=initrd.lz autoinstall url=${base}/ubuntu1804.iso net.ifnames=0 ip=dhcp ipv6.disable=1 ds=nocloud-net;s=${seedfrom}
initrd ${base}/initrd.lz
boot

**Reply by sebaxakerhtc** [reply]
Oh, god :)
```
set seedfrom http://192.168.140.21/iso/ubuntu1804/
set base http://192.168.140.21/iso/ubuntu1804

kernel ${base}/vmlinuz initrd=initrd.lz autoinstall url=${base}/ubuntu1804.iso net.ifnames=0 ip=dhcp ipv6.disable=1 ds=nocloud-net;s=${seedfrom}
initrd ${base}/initrd.lz
boot
```

**Reply by jamaya77** [reply]
HI @sebaxakerhtc ,

Thanks again your time. Unfortunately there was no change.

#!ipxe
#ubuntu 18.04

#$seedfrom used to find the user-data and meta-data files by nocloud-net provider for cloud-init.
#the trailing slash is important, the cloud-init sticks 'meta-data' or 'user-data' right after it, without prepending a forward slash.
set seedfrom http://192.168.140.21/iso/ubuntu1804/

#$base url is where the vmlinuz and initrd live.
#they were fished out from the live-server iso file. the iso file is also in this directory.
set base http://192.168.140.21/iso/ubuntu1804

kernel ${base}/vmlinuz initrd=initrd.lz autoinstall url=${base}/ubuntu1804.iso net.ifnames=0 ip=dhcp ipv6.disable=1 ds=nocloud-net;s=${seedfrom}
initrd ${base}/initrd.lz
boot

**Reply by sebaxakerhtc** [reply]
Hi.
That means you don't have necessary files there (autoisnall, initrd, etc.) 
Maybe, just iso... It's not correct.

**Comment by jamaya77** [comment]
HI @sebaxakerhtc 

Just wanted to inform you that I found and tested a solution which seems to work well. Although I'm no longer deploying the actual Ubuntu 18.04 Desktop ISO, I created alternative network installer [solution](https://noobient.com/2019/06/25/ubuntu-18-04-uefi-network-installer/) which deploys as a mini.iso with the option to install "Ubuntu-Desktop". It's much faster and it configures NetPlan and Network-Manager automatically to dhcp. Supports both UEFI and BIOS based machines.

#!ipxe
#ubuntu 18.04
#uefi

kernel http://192.168.140.21/iso/ubuntu1804/linux ||
initrd http://192.168.140.21/iso/ubuntu1804//initrd.gz ||
imgargs linux initrd=initrd.gz vga=791 toram
imgstat
boot || read void
goto start

**Reply by sebaxakerhtc** [reply]
That nice! Congratulations 👏
For my purposes i use net-install. 
But if there is no internet access – your solution may be helpful
